gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -358,9 +357,7 @@ public WritableByteChannel create(final StorageResourceId resourceId, CreateObje
      */
 
     Optional<Long> overwriteGeneration =
-        resourceId.hasGenerationId()
```

Comment:
I think that we may need to preserve `resourceId.hasGenerationId()` logic so caller can control if they want to overwrite a specific generation and fail on concurrent writes, or always overwrite regardless of concurrent overwrites. Does any code in connector depends on this logic? Does connector sets generation in `resourceId` parameter at any time?

Reply:
Per discussion this will be kept.